### DISCUSSION TITLE
More frequent releases for enhancement/bug fix?

### DISCUSSION THREAD

**Discussion by pengchen0692** [fo-dicom/fo-dicom#1887] [General]
**What is, according to you, missing from fo-dicom? Please describe.**
_Context:_
It's regarding issue [JsonDicomConverter allows serializing DS/IS dicom item with invalid values](https://github.com/fo-dicom/fo-dicom/issues/1355), which should be fixed by PR #1354 soon. But seems next release of fo-dicom would not be very soon based on [release history](https://github.com/fo-dicom/fo-dicom/releases), which mean we won't benefit from the fix until next release.  I'm wondering if there is any way to solve it. 

**Describe the solution you'd like**
I have 2 proposals in mind.
1.  Change fo-dicom to release frequently -- but I believe there is reason underneath for current release pace,  so this probably not a good idea.
2.  We build our own infrastructure to build private binaries, and publish to our private nuget. In this way, we could benefit from fix immediately and also not bother normal fo-dicom releases. 

I personally prefer the 2nd one, but  I have no idea about how fo-dicom is published. So if there is any guideline for that, would be very helpful!

**Describe possible alternatives or existing workarounds you've considered**
N/A

**Comment by amoerie** [comment]
The reason for the "slow" release cadence is mostly historical I believe, since fo-dicom used to be a lot more inactive. (And compared to other open source projects, it probably still is very inactive).

Furthermore, the "keys to the castle" are currently in the sole hands of @gofal, who is the only one that is able to publish to nuget.
That is also why I'm not merging any open pull requests yet, because they cannot be published yet, so I'd rather have them be reviewed by @gofal as well if we have to wait anyway.

Since @gofal has been absent here for a while now, I understand this might be annoying for you to wait so long for another release. 

I personally use a forked fo-dicom as a git submodule, which allows me to make changes in my fork and use them directly, while retaining the ability to merge upstream changes back into my fork. It also simplified stepping into the code while debugging.
The downside is that I am always building fo-dicom alongside my application code in continuous integration.

As for the packaging, I think @gofal uses Setup/0_BuildAndPack5.cmd, but I'm not sure whether it's up to date...

**Comment by mrbean-bremen** [comment]
It probably makes sense to add instructions how to create a local package somewhere (Wiki or documentation), this should not be too much.

**Comment by pengchen0692** [comment]
The instructions should be very helpful. :)
Thanks for `Setup/0_BuildAndPack5.cmd`, it should be a good start for me!

**Comment by jaime-olivares** [comment]
It seems the question has been already replied with an explanation about the release cadence.
